lib/_stream_writable.js

```diff
@@ -134,11 +134,28 @@ Object.defineProperty(WritableState.prototype, 'buffer', {
      'instead.')
 });
 
+// Test _writableState for inheritance to account for Duplex streams,
+// whose prototype chain only points to Readable.
+var realHasInstance;
+if (Symbol.hasInstance) {
+  realHasInstance = Function.prototype[Symbol.hasInstance];
+  Object.defineProperty(Writable, Symbol.hasInstance, {
+    value: function(object) {
+      return object._writableState instanceof WritableState;
+    }
+  });
+} else {
+  realHasInstance = function(object) {
+    return object instanceof this;
+  };
+}
+
 function Writable(options) {
-  // Writable ctor is applied to Duplexes, though they're not
-  // instanceof Writable, they're instanceof Readable.
-  if (!(this instanceof Writable) && !(this instanceof Stream.Duplex))
+  // Writable ctor is applied to Duplexes, too.
+  if (!(realHasInstance.call(Writable, this)) &&
+      !(this instanceof Stream.Duplex)) {
```

Comment:
> I think that will reduce our cruft.

@mcollina Yeah, me too, so I tried it – that’s what breaks `LazyTransform`, because it invokes the `Writable` constructor from inside a `_writableState` getter. :(